TruePath/PathExtensions.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace TruePath;
+/// <summary>
+/// Extension methods for <see cref="IPath"/> and <see cref="IPath{TPath}"/>.
+/// </summary>
+public static class PathExtensions
+{
+    /// <summary>
+    /// Gets the extension of the file name of the path.
+    /// </summary>
+    /// <typeparam name="TPath"></typeparam>
+    /// <param name="path"></param>
+    /// <returns></returns>
+    public static string GetExtension(this IPath path)
+    {
+        var fileExtenstion = Path.GetExtension(path.FileName);
+        if (string.IsNullOrEmpty(fileExtenstion))
+        {
+            throw new ArgumentException(string.Format("{0} is not a file", path.FileName));
+        }
+        return fileExtenstion;
+    }
+    /// <summary>
+    /// Gets the extension of the file name of the path without the dot.
+    /// </summary>
+    /// <typeparam name="TPath"></typeparam>
+    /// <param name="path">
+    /// The path to get the extension from.
+    /// </param>
+    /// <returns>The extension of the file name of the path without the dot.</returns>
+    public static string GetExtensionWithoutDot<TPath>(this TPath path) where TPath : IPath
```

Comment:
`<TPath>` is not required here: you can make it to accept `IPath` as well.

Also, let's make the same amendments as above, and cover with tests:
- `null` for file with no extension,
- empty string for file whose name ends with dot.